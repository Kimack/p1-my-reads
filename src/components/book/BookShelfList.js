import React from 'react'
import BookShelf from './BookShelf.js'

const BookShelfList = ({ shelves, books, onBookShelfChange }) => (
    <div className="list-books-content">
            <ul>
                {shelves.map((shelf) => {
                    return (<ol key={shelf.id}>
                        <BookShelf
                            shelf={shelf}
                            books={
                                books.filter((book) => {
                                    return book.shelf === shelf.id
                                })
                            }
                            shelves={shelves}
                            onBookShelfChange={onBookShelfChange}
                        />
                    </ol>)
                })}
            </ul>
    </div>
)

export default BookShelfList;