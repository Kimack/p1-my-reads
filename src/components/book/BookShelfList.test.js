import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import BookShelfList from './BookShelfList';

configure({ adapter: new Adapter() });

describe('BookList instance is rendered corretly', () => {
    it('renders without crashing', () => {
        const component = <BookShelfList books={[{ id: 0 }, { id: 1 }]} shelves={[{id: 0}, {id: 1}]} onBookShelfChange={() => { }} />;
        expect(shallow(component)).toMatchSnapshot();
    });
});