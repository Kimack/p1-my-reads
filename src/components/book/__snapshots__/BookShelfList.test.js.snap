// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookList instance is rendered corretly renders without crashing 1`] = `
<div
  className="list-books-content"
>
  <ul>
    <ol
      key="0"
    >
      <BookShelf
        books={Array []}
        onBookShelfChange={[Function]}
        shelf={
          Object {
            "id": 0,
          }
        }
        shelves={
          Array [
            Object {
              "id": 0,
            },
            Object {
              "id": 1,
            },
          ]
        }
      />
    </ol>
    <ol
      key="1"
    >
      <BookShelf
        books={Array []}
        onBookShelfChange={[Function]}
        shelf={
          Object {
            "id": 1,
          }
        }
        shelves={
          Array [
            Object {
              "id": 0,
            },
            Object {
              "id": 1,
            },
          ]
        }
      />
    </ol>
  </ul>
</div>
`;
